state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java

```diff
@@ -28,25 +29,34 @@
 @RequiredArgsConstructor
 public class DefaultApolloClientFactory implements ApolloClientFactory {
   @NonNull private final String streamRegistryUrl;
-  @NonNull private final Consumer<ApolloClient.Builder> configurer;
+  private Credentials credentials;
 
-  public DefaultApolloClientFactory(String streamRegistryUrl) {
-    this(streamRegistryUrl, builder -> {});
+  public DefaultApolloClientFactory(String streamRegistryUrl, Credentials credentials) {
+    this(streamRegistryUrl);
+    this.credentials = credentials;
   }
 
   @Override
   public ApolloClient create() {
-    var builder = builder()
-        .okHttpClient(new OkHttpClient.Builder().build());
-    configurer.accept(builder);
-    return builder
+    var okHttpClientBuilder = new OkHttpClient.Builder();
+    if (credentials != null) {
+      okHttpClientBuilder.addInterceptor(chain -> chain.proceed(chain.request().newBuilder()
+          .header("Authorization", (Base64.getEncoder().encodeToString((credentials.getUserName() + ":" + credentials.getPassword()).getBytes(UTF_8))))
```

Comment:
Doesn't need the braces around (Base64.....)

Reply:
Are we sure this is the expected "Authorization" value. I think the canonical form is
"Basic <thebytes>", eg: for 'user' and 'pass', the Authorization would be "Basic dXNlcjpwYXNz"

I could be wrong, but would be good to get this confirmed and documented.

Reply:
Perhaps extract to a BasicAuthEncoder and add an illustrative test.

Reply:
yes, it needs to be "Basic dXNlcjpwYXNz". will fix that

Reply:
fixed